### ISSUE TITLE
[Feature Request] Set colour to colour of menu bar item selected by OS

### ISSUE THREAD

**Issue by bnice** [yujitach/MenuMeters#204]
Hi,

With Big Sur making the menu bar being more transparent and a dynamic background (changes for day and night) selected. I see the other menu items such as battery, spotlight, and control  change from black for the light (day) background and black for the dark (night) background. I see the "Magnet" menu bar item does this so there must be an API for this. Perhaps set this is the default and let the user override this by setting the colour to their own choice? But then you want the user to choose to reset to this option after they have select a colour if they change their mind.

Lastly, I would like to thank you for continuing to develop and support Menu Meters 🙂. I can't imagine OS X without I have had it for so many years.

Regards Ben.

**Comment by yujitach** [comment]
Yes it’s the default for monochrome images, but many “meters” require more than one color. What would you suggest MenuMeters to do in such cases? 

Also, I think most of the texts should automatically follow the system default already...  Which elements in MenuMeters do you want me to deal with?

**Comment by bnice** [comment]
Hi,

Basically every where the user can choose a colour, you need to allow the user select the system default colour. I take the system default is the one that changes colour with light and dark backgrounds?  I saw after I posted this someone else has requested the same feature. I saw you asking if they had chosen black for the colour? Is this because if the user selects black they actually get the system default colour that changes with light and dark backgrounds? If this is so why not do make black do this in all places the user can select a colour? 

I guess the colour selection dialog you are bring up for the user is built in OS X? So, you can't add in anyway a way to allow the user to select the system default colour? 

Ben.

**Comment by yujitach** [comment]
What's discussed here is definitely a necessary addition. Also discussed in a previous issue #88 was:
- auto invert the color
- enhance the pref pane UI so that the colors for the light mode and the dark mode can be chosen separately.

**Comment by yujitach** [comment]
I was told in an email that there are also system colors that automatically adapts to the light/dark mode, as described in https://developer.apple.com/design/human-interface-guidelines/macos/visual-design/color/ . 

And they can be already chosen within the standard color palette, although it is not easily discoverable:

![Screen Shot 2020-12-16 at 10 10 45](https://user-images.githubusercontent.com/5449811/102292225-1546dd80-3f88-11eb-9627-106284783f23.png)

And they do look good on both the dark and light menu bar.

**Comment by bnice** [comment]
Hi,

Thanks, I will give them a try.

Ben.

On December 16, 2020 at 2:19 PM, yujitach <notifications@github.com> wrote:


I was told in an email that there are also system colors that automatically adapts to the light/dark mode, as described in https://developer.apple.com/design/human-interface-guidelines/macos/visual-design/color/ .
And they can be already chosen within the standard color palette, although it is not easily discoverable:

And they do look good on both the dark and light menu bar.

**Comment by radome** [comment]
System colors are nice but when the dark/light shift is high, they're not performing very well IMO.

I found that "LabelColors" (see image below) actually switch between dark and light with the rest of the menu bar. 
![Screenshot 2021-01-06 at 17 19 50](https://user-images.githubusercontent.com/1145093/103800006-bdaa0800-5043-11eb-8193-3e56e2ad492f.png)

In the two images below, `quaternaryLabelColor` and `tertiaryLabelColor` with both dark and light background.
![Screenshot 2021-01-06 at 17 20 20](https://user-images.githubusercontent.com/1145093/103800038-ca2e6080-5043-11eb-9879-d0f14943dbf2.png)

![Screenshot 2021-01-06 at 17 20 41](https://user-images.githubusercontent.com/1145093/103800059-d3b7c880-5043-11eb-9796-64fd62824e88.png)

Undortunately those are only shades of grey, but I'm not sure if would be feasible to create system colors that automatically switch between dark and light...

Anyway, for the sake of this issue, I think it would be nice if the user could select which color to apply when the menu bar in on a dark or light background (although I don't know if that would be possible)

**Comment by yujitach** [comment]
Yeah. And what makes the problem worse on Big Sur is that the menu bar color actually depends on the desktop background behind...

>I think it would be nice if the user could select which color to apply when the menu bar in on a dark or light  background (although I don't know if that would be possible)

It's technically possible. But coming up with a sensible UI to achieve that is more difficult. (For example, I don't want to double the number of the color wells in the preference window...)

**Comment by bnice** [comment]
Hi,

I found SystemRedColor and SystemGreenColor okay substitutes for the red and green on the memory and network widgets definitely an improvement. Likewise SystemBlueColor, SystemRedColor, and SystemOrangeColor work well the cpu graphs.

It hard battle for colour though, it looks like Apple are aiming for only black and white in the menu bar, as all their items appear to be.

**Comment by h4wwk3ye** [comment]
Hi, 
Is there any update on this. I use Irvue to change my walllpapers every 1 hour and none of the colors in developer section work fine always, I only use the network monitor. Can we get an option to just set it to the color selected by other system icons in big sur?

**Comment by alasdairhurst** [comment]
Dark mode with light background is pretty much unusable
![image](https://user-images.githubusercontent.com/8441810/122241463-7a897600-ceba-11eb-82a5-0c7cdbe2fa1a.png)

**Comment by navinthomas** [comment]
If you simply need an alternative black/white colour, just like how the other menubar icons and text works, then choose `textColor` in Developer section. It works extremely well when dark mode is switched on or off.

![image](https://user-images.githubusercontent.com/34918736/163759080-7f4b1450-efa7-4bb3-b487-bfbf631efdbf.png)

**Comment by melyux** [comment]
4 years later, crickets.
